### PR TITLE
Fix SPDY fails in node >= 11.1.0

### PR DESCRIPTION
**Pull request by yi-ge**
In node v11+:
```
util.js:305
    throw new ERR_INVALID_ARG_TYPE('superCtor.prototype',
    ^

TypeError [ERR_INVALID_ARG_TYPE]: The "superCtor.prototype" property must be of type Object. Received type undefined
```

- [x] This is a **bugfix**
- [ ] This is a **feature**
- [ ] This is a **code refactor**
- [ ] This is a **test update**
- [ ] This is a **docs update**
- [ ] This is a **metadata update**

### CODE REVIEW

**Comment by yi-ge** [comment]
About it: `https://github.com/spdy-http2/node-spdy/issues/350`

File: lib/Server.js [hunk 1]
```diff
@@ -630,7 +629,9 @@ function Server(compiler, options = {}, _log) {
     if (semver.gte(process.version, '10.0.0')) {
       this.listeningApp = https.createServer(options.https, app);
     } else {
-      this.listeningApp = spdy.createServer(options.https, app);
+      /* eslint-disable global-require */
+      this.listeningApp = require('spdy').createServer(options.https, app);
+      /* eslint-enable global-require */
```
Comment (evilebottnawi) [inline lib/Server.js]: Not sure it is solve the problem

**Comment by evilebottnawi** [comment]
Also please accept CLA

**Comment by yi-ge** [comment]
**t.js:**
```
const spdy = require('spdy');
```

You can run it (`node t.js`) in node v11.9.0.
image

About: [https://github.com/spdy-http2/node-spdy/issues/350](https://github.com/spdy-http2/node-spdy/issues/350)

**Comment by yi-ge** [comment]
😄

**Comment by evilebottnawi** [comment]
@yi-ge looks you catch the bug in nodejs :smile: Can you create issue in `nodejs` and add comment with link in source code

**Comment by yi-ge** [comment]
@evilebottnawi Thank you.

**Comment by evilebottnawi** [comment]
Strange can't reproducible locally on latest nodejs

**Comment by yi-ge** [comment]
@evilebottnawi sdpy v4.0？

Node.js 11.9.0, spdy 4.0.

**Comment by evilebottnawi** [comment]
@yi-ge 
> Now using node v11.9.0 (npm v6.5.0)

can we add tests for this case and looks what CI show

**Comment by yi-ge** [comment]
@evilebottnawi It's just a line of code.

**Comment by yi-ge** [comment]
image
vue cli create project.

**Comment by yi-ge** [comment]
https://github.com/webpack/webpack-dev-server/issues/1592

**Comment by evilebottnawi** [comment]
@yi-ge Strange, can't reproduce:
```
node -v
v11.9.0
```

Input:
```
const spdy = require('spdy');

console.log(spdy);
```

Output:
```
node test.js

{ handle: { [Function: Handle] create: [Function: create] },
  request: { onNewListener: [Function: onNewListener] },
  response:
   { writeHead: [Function: writeHead],
     end: [Function: end],
     push: [Function: push],
     writeContinue: [Function: writeContinue] },
  Socket: [Function: Socket],
  agent:
   { Agent: { [Function: Agent] create: [Function: create] },
     PlainAgent: { [Function: Agent] create: [Function: create] },
     create: [Function: create] },
  Agent: { [Function: Agent] create: [Function: create] },
  createAgent: [Function: create],
  server:
   { Server: { [Function: Server] create: [Function: create] },
     PlainServer: { [Function: Server] create: [Function: create] },
     create: [Function: create] },
  Server: { [Function: Server] create: [Function: create] },
  PlainServer: { [Function: Server] create: [Function: create] },
  createServer: [Function: create] }

```

What os you use?

**Comment by yi-ge** [comment]
@evilebottnawi MacOS. Merging code does not affect the result. You can first merge and verify. 😄

**Comment by evilebottnawi** [comment]
@yi-ge So this PR doesn't solve problem?

**Comment by yi-ge** [comment]
@evilebottnawi This OR can solve the problem.

**Comment by yi-ge** [comment]
@evilebottnawi This PR, No consequences. 😄

**Comment by evilebottnawi** [comment]
@yi-ge i can't understand you, i can't reproduce problem :disappointed: Maybe it is happens only on `macos`?

**Comment by yi-ge** [comment]
image

**Comment by evilebottnawi** [comment]
@yi-ge looks like all works fine, i can't understand problem, please clarrify

**Comment by yi-ge** [comment]
@evilebottnawi The same code cannot be run in Mac OS node v11.1.0 +. You can see: [https://github.com/spdy-http2/node-spdy/issues/350](https://github.com/spdy-http2/node-spdy/issues/350) or [https://github.com/webpack/webpack-dev-server/issues/1592](https://github.com/webpack/webpack-dev-server/issues/1592). This PR avoid this kind of situation from happening.

**Comment by evilebottnawi** [comment]
@yi-ge thanks for clarify, just add comment in code with link on issue and we can merge this :+1:

**Pull request merged**